functions/nut.bash

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env bash
+# shellcheck disable=SC2154,SC2155
+
+## Check if Network UPS Tools is installed
+##
+##  nut_is_installed()
+##
+nut_is_installed() {
+  if dpkg -s 'nut' &> /dev/null; then return 0; fi
+  return 1
+}
+
+## (Un)Install Network UPS Tools
+## Supports both INTERACTIVE and UNATTENDED mode
+##
+## Valid arguments: "install" or "remove"
+##
+##   nut_install(String action)
+##
+nut_install() {
+  if [[ $1 == "remove" ]]; then
+    if [[ -n $INTERACTIVE ]]; then
+      whiptail --title "nut removal" --msgbox "This will remove Network UPS Tools (nut)." 7 80
+    fi
+    echo -n "$(timestamp) [openHABian] Removing Network UPS Tools... "
+    if cond_redirect apt purge -y nut; then echo "OK"; else echo "FAILED"; return 1; fi
+    return;
+  fi
+
+  if [[ $1 != "install" ]]; then return 1; fi
+  if [[ -n $INTERACTIVE ]]; then
+    whiptail --title "nut installation" --msgbox "This will install Network UPS Tools (nut)." 8 80
+  fi
+
+  echo -n "$(timestamp) [openHABian] Installing Network UPS Tools... "
+  if cond_redirect apt install -y nut; then echo "OK"; else echo "FAILED (package installation)"; return 1; fi
+
+  ## Do service activation later as nut-server & nut-client depend on the setup!
+}
+
+## Configure Network UPS Tools
+## Supports both INTERACTIVE and UNATTENDED mode
+##
+## Variables used:
+##   - "nutmode": "netserver" or "netclient"
+##   for "netserver":
+##   - "nutupsdriver": driver for local UPS
+##   - "nutupsdescr": description for local UPS
+##  for "netclient":
+##   - "nutupsname": name of network UPS
+##   - "nutupshost": host of network UPS
+##   - "nutupsuser": username for network UPS
+##   - "nutupspw": password for network UPS
+##
+##   nut_setup()
+##
+nut_setup() {
+  if [[ -n $UNATTENDED ]] && [[ -z $nutmode ]]; then
+    echo "$(timestamp) [openHABian] Beginning Network UPS Tools setup... SKIPPED (no configuration provided)"
+    return 0
+  fi
+
+  local upsmasterpw="$(openssl rand -base64 12)"
+  local upsslavepw="$(openssl rand -base64 12)"
+  local upsmonitor="MONITOR ups@localhost 1 upsmaster ${upsmasterpw} master"
+
+  local whiptailTitle="Network UPS Tools"
+  local introText="Network UPS Tools (nut) adds an Uniterruptable Power Supply (UPS) to your openHABian system.\\nIn case of a power outage, it protects your system from data loss by ensuring a safe shutdown.\\n\\nWe will guide you through the setup."
+
+  if ! nut_is_installed; then nut_install "install"; fi
```

Comment:
@mstormi Here it will install nut if it is not installed. I have tested this as well, works 👍